packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3311,10 +3311,35 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   @override
   void initState() {
     super.initState();
-    assert(
-      widget.pages.isEmpty || widget.onPopPage != null,
-      'The Navigator.onPopPage must be provided to use the Navigator.pages API',
-    );
+    assert((){
+      if (widget.pages != const <Page<dynamic>>[]) {
+        // This navigator uses page API.
+        if (widget.pages.isEmpty) {
+          FlutterError.reportError(
```

Comment:
What's the difference between FlutterError.reportError and using `throw`?  I guess the result is the same since your tests are passing.

Reply:
I think devtool insert some hook in reportError, using reportError maybe a better choice if we know the error is caused by the user.